### ISSUE TITLE
"Improvement in the German translation" or "Warning when small buildings are tagged as 'building=terrace'".

### ISSUE THREAD

**Issue by AndreasHoffmann2** [streetcomplete/StreetComplete#3793]
Hi,

yesterday, I tagged a bunch of houses as "building=terrace". As I learned today, this tag is meant for a whole row of houses. And not for the individual houses in that row.

In the German translation, the label for this is "Reihenhäuser", the plural of "rowhouse". Strictly speaking, the description is 100% correct. I misunderstood it anyway, since it did not even occur to me, that it is possible to tag a whole row of houses as a single building. This might be obvious for experienced users. For me, it was not. And since StreetComplete can and should be used by users with less experience, there is room for improvement IMO.

Thus I would like to suggest two possible solutions for this:
1.) The German description could make it clear that this is not meant for individual houses.
2.) There could be a warning, when the user wants to tag a building with an area of less than e.g. 100 m² as "building=terrace".

The latter one would solve the issues for all languages at once, while the first one does only fix the issue for German users.

**Comment by matkoniecz** [comment]
Translations can be changed by anyone (see https://github.com/streetcomplete/StreetComplete/blob/master/CONTRIBUTING.md#translating-the-app ) but it may be in practice request to change also base English version.

**Comment by FloEdelmann** [comment]
See also #1973.

**Comment by westnordost** [comment]
The icon already shows several houses and the description of "Reihenhäuser" already says "eine lange Reihe aus ähnlichen Einfamilienhäusern", I think this is clear enough. Besides, the mistagging you made is really really minor, in my opinion. About as minor as tagging `building=semi_detached` when it should be `building=house` instead (or the other way round) and in any case is an improvement in detail from `building=yes`.

**Comment by westnordost** [comment]
Still, if you have an idea to make the description even clearer (without making it too long), see @matkoniecz , you can contribute by enhancing the German translation.

**Comment by AndreasHoffmann2** [comment]
I don't agree, that this is clear enough. To check this, I queried for "building=terrace" in Berlin with a timestamp after 2021-09-01. Most usages I found are wrong _and_ created by StreetComplete.

I did also post a query about how to fix my edits in the German OSM-forum. And immediately, at two other users reacted that made the same mistake. It's not just me.

The current description is only clear enough if you know that it is a possibility to tag a whole row of houses as a single building. I don't think that this can be assumed to be common knowledge. If you don't know that, tagging row-houses as row-houses seems perfectly fine. The only hint, StreetComplete gives to the user, is the usages of the plural-form instead of singular. That is not enough for a user-interface that should keep the user from misunderstanding ist.

And the amount of mistakes introduced by StreetComplete does support that.

**Comment by matkoniecz** [comment]
Is "Multiple rowhouses" still going to be fit and be significantly more clear?

**Comment by westnordost** [comment]
I don't understand what would be misunderstable with "eine lange Reihe aus ähnlichen Einfamilienhäusern".

**Comment by AndreasHoffmann2** [comment]
> I don't understand what would be misunderstable with "eine lange Reihe aus ähnlichen Einfamilienhäusern".

If you take it out of the context, the text is perfectly fine and correct. It's the context that can throw the user off. Most users have never seen multiple houses tagged as a single object in OSM. 
Now, StreetComplete offers him a long list of options to classify a single building. All housing-options in this list do tag a single building only. And there are special options for houses that are detached and semi-detached. But no explicit option for a single "rowhouse".

Now the user sees the option "rowhouses". What does he probably think?

Option 1: There is the option, I was looking for.
Option 2: This must be the only only option that tags multiple houses as one. Even though I have never seen that on OSM.

In my head, it was option 1. The fit was too good for the question I faced. I actually had the feeling that something was not 100% correct due to the explanatory text. But hey, not all texts you see everyday are 100% precise (no offense!).

As for the improvement of the text: I don't have a good idea currently. For me, something like "multiple rowhouses within one OSM-object" would have helped. But I don't like the technicality in this.

**Comment by westnordost** [comment]
As written, to be honest I don't see a big issue with tagging a single row house as "rowhouses", the mistake is minimal and I wouldn't know a data consumer where the difference mattered. It would not surprise me at all if `building=terrace` would be used **a lot** also for single row houses, also before StreetComplete came along. (See the echo in that forum thread. They are not all StreetComplete users.)

It's the same issue with `building=semi_detached` really. According to the wiki, `semi_detached` should be used for **each side** of a duplex house but many duplex homes are mapped as one building (IMO rightly so, because they share one roof). As a consequence, many `building=semi_detached` you'll find in the wild are tagged "wrong" because the whole outline is tagged as such.

Alternatively, we could accept the fact that this is the de-facto tagging situation, i.e. that `building=semi_detached` is a duplex home **or** one of two parts of a duplex home. I think this is what data consumers need to assume anyway. Same with `terrace`. Both of these ~mistakes can easily and quickly be spotted from aerial imagery and corrected in JOSM by people who want to have this 100% correct, for whatever reason.

But the important and most time-consuming part of this has already been done - the surveying what kind of building this is. Without this survey, a JOSM mapper could not even know whether something is a detached house, a semidetached house, several houses sharing some walls, an apartment building or even something else. So, even if the resulting tagging is "wrong", it still is a benefit because 
1. it records the type of building (without destroying data)
2. even if wrong, it adds information on the type of `house`: e.g. it could be retagged to `building=house` + `house=terraced` or whatever

**Comment by westnordost** [comment]
By the way, in recent years, the `house=*` has gained some traction (mostly in the United Kingdom):

https://wiki.openstreetmap.org/wiki/Key:house

**Comment by westnordost** [comment]
> make a tagging proposal to seek approval for [building=terraced_house](https://taginfo.openstreetmap.org/tags/building=terraced_house) (used 3 567 as of writing this post) to tag a single terraced house and add it to StreetComplete afterwards

https://taginfo.openstreetmap.org/tags/house=terraced has 30 times more usages.

**Comment by delexi** [comment]
I am another user who was bitten by this and who fixed their tagging after seeing the forum post mentioned above.

I think explicitly excluding a single Reihenhaus from the building=terrace answer is a good idea. It makes it clearer without a lot of extra text.

Additionally I vote for not putting Einfamilienhaus and Reihenhaus in the same answer, but instead have one answer for an Einfamilienhaus (building=detached) and another one for a Reihenhaus (building=house + house=terraced).

**Comment by AndreasHoffmann2** [comment]
I like the idea of having two answers:
1) rowhouses: "a whole row of rowhouses" (building=terrasse)
2) rowhouse: "one house in the row of houses" (icon: the one for multiple rowhouses but zoomed so far that only 1/3 of the neighbours is visible; tags: building=house, house=terraced)

This would remove the ambiguity and improve the quality of the data recorded as well.

**Comment by westnordost** [comment]
I reckon you want to give the most detailed answer possible, but note that `building=*` is a "Fass ohne Boden" in this regard because this key is (still) treated as somewhat of tag with free-form value. 
Theoretically there is an almost infinite number of possible values to support. But answering the quest should still be reasonably quick, so only the most important ones are displayed. So, I don't see a reason to include a separate "single row house" to differenciate to "single house". 

However, what I'll do is to change the description of `building=house` to "non-detached single-family home, e.g. a single row house" so that it is clear that it also includes single row houses.
